src/yunohost/tools.py

```diff
@@ -555,3 +556,22 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+
+def tools_portavailable(port):
+    """
+    Check availability of a local port
+
+    Keyword argument:
+        port -- Port to check
+
+    """
+    try:
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.settimeout(1)
+        s.connect(("localhost", int(port)))
```

Comment:
I think the idea of this helper is mainly to check if a port is available *on the local machine*, not to test if a port is reachable from outside (which is a much more complex issue since anyway you need to ask another machine, which is what is done with ports.yunohost.org). Though it would definitely be interesting to have such a helper, imho it's not the same semantic (should be 'portreachable' or something like that). (Also : that would assume that the port is open in the firewall)